src/services/TransactionCommand.ts

```diff
@@ -191,13 +194,53 @@ export class TransactionCommand {
 
     private resolveFeeMultipler(transaction: Transaction): number | undefined {
         if (transaction.maxFee.compact() == 1) {
-            return this.transactionFees.medianFeeMultiplier || this.networkConfiguration.defaultDynamicFeeMultiplier;
-            // TODO uncomment the following line when https://github.com/nemtech/catapult-rest/issues/326 is resolved
-            // return this.transactionFees.averageFeeMultiplier * 1.2 || this.networkConfiguration.defaultDynamicFeeMultiplier;
+            const fees =
+                this.transactionFees.medianFeeMultiplier < this.transactionFees.minFeeMultiplier
+                    ? this.transactionFees.minFeeMultiplier
+                    : this.transactionFees.medianFeeMultiplier;
+            return fees || this.networkConfiguration.defaultDynamicFeeMultiplier;
         }
         if (transaction.maxFee.compact() == 2) {
-            return this.transactionFees.highestFeeMultiplier || this.networkConfiguration.defaultDynamicFeeMultiplier;
+            const fees =
+                this.transactionFees.highestFeeMultiplier < this.transactionFees.minFeeMultiplier
+                    ? this.transactionFees.minFeeMultiplier
+                    : this.transactionFees.highestFeeMultiplier;
+            return fees || this.networkConfiguration.defaultDynamicFeeMultiplier;
         }
         return undefined;
     }
+
+    public announceSimpleLedger(
+        service: TransactionAnnouncerService,
+        signedTransactions: Observable<SignedTransaction>[],
+    ): Observable<BroadcastResult>[] {
+        return signedTransactions.map((o) => o.pipe(flatMap((s) => service.announce(s))));
+    }
+
+    public announceHashAndAggregateBondedLedger(
+        service: TransactionAnnouncerService,
+        signedTransactions: Observable<SignedTransaction>[],
+    ): Observable<BroadcastResult> {
+        return signedTransactions[0].pipe(
+            flatMap((signedHashLockTransaction) => {
+                return signedTransactions[1].pipe(
+                    flatMap((signedAggregateTransaction) => {
+                        return service.announceHashAndAggregateBonded(signedHashLockTransaction, signedAggregateTransaction);
+                    }),
+                );
+            }),
+        );
+    }
+
+    public calculateSuggestedMaxFeeLedger(transaction: Transaction): Transaction {
```

Comment:
@TriPhan96  could have turned the already existed [calculateSuggestedMaxFee()](https://github.com/nemgrouplimited/symbol-desktop-wallet/blob/d19854bb694716f71961606fa69fce0b204651b6/src/services/TransactionCommand.ts#L180 ) function into a public one and re-consumed it, as the newly created one for ledger is containing same logic which leads to code redundancy.

Reply:
Okay, i will update it based on your recommendation

Reply:
I have used the same way for [announceSimpleLedger()](https://github.com/FDS-Ledger/symbol-desktop-wallet/blob/13982f675c0c1f7486061fea553c06860df7a1cd/src/services/TransactionCommand.ts#L213) and [announceHashAndAggregateBondedLedger()](https://github.com/FDS-Ledger/symbol-desktop-wallet/blob/13982f675c0c1f7486061fea553c06860df7a1cd/src/services/TransactionCommand.ts#L220). Should I change it that same way as `calculateSuggestedMaxFee()` function?

Reply:
Yes, it will be better.